File: tests/uasm/sample/diamond_simp/DiamondSimpTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import type { UdonInt } from "@ootr/udon-assembly-ts/stubs/UdonTypes";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class DiamondSimpTest extends UdonSharpBehaviour {
  private score: UdonInt = 0 as UdonInt;

  Start(): void {
    this.score = 75 as UdonInt;
    // biome-ignore lint/complexity/noUselessTernary: intentional — tests diamond simplification of `cond ? true : false` pattern
    const passed: boolean = this.score >= (60 as UdonInt) ? true : false;
    // biome-ignore lint/complexity/noUselessTernary: intentional — tests diamond simplification of `cond ? true : false` pattern
    const failed: boolean = this.score < (60 as UdonInt) ? true : false;
    Debug.Log(passed);
    Debug.Log(failed);
  }
}

File: tests/uasm/sample/diamond_simp/DiamondSimpTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class DiamondSimpTest : UdonSharpBehaviour
{
    private int score;

    public void Start()
    {
        score = 75;
        bool passed = score >= 60 ? true : false;
        bool failed = score < 60 ? true : false;
        Debug.Log(passed);
        Debug.Log(failed);
    }
}
